### PR TITLE
feat: add content_sid variable

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

Supports: https://github.com/supabase/infrastructure/pull/15123 (addenum PR with small fix: https://github.com/supabase/infrastructure/pull/15439)

Tested locally.

### CODE REVIEW

File: apps/docs/pages/guides/auth/phone-login/twilio.mdx [hunk 1]
```diff
@@ -400,11 +406,16 @@ Twilio Verify can be used with WhatsApp OTPs with no additional configuration.
 Complete the following steps to use WhatsApp OTP with Twilio Programmable Messaging:
 
 1. Go through the [Twilio self sign up guide for WhatsApp](https://www.twilio.com/docs/whatsapp/self-sign-up).
-2. Follow the [Twilio Guide for Submitting Templates](https://www.twilio.com/docs/whatsapp/tutorial/send-whatsapp-notification-messages-templates#creating-message-templates-and-submitting-them-for-approval) and submit a Basic Authentication Message for approval (e.g. Your Code is 123456).
+2. Follow the [Twilio Guide for Submitting Templates](https://www.twilio.com/docs/whatsapp/tutorial/send-whatsapp-notification-messages-templates#creating-message-templates-and-submitting-them-for-approval) and submit a Basic Authentication Message for approval.
+
+Alternatively, you can register a Twilio Content SID which corresponds to a message template.
+![Twilio Content SID Image](/docs/img/guides/auth-twilio/twilio_content_sid.png)
+
+Note that you may only use one Twilio Content SID at a time. Consequently, we recommend using Twilio Verify or the Basic Authentication Message if you require internationalization on your message template.
```
Comment (saltcod) [inline apps/docs/pages/guides/auth/phone-login/twilio.mdx:414]: maybe don't need "consequently" here

File: apps/docs/pages/guides/auth/phone-login/twilio.mdx [hunk 2]
```diff
@@ -114,6 +114,12 @@ Note: for "Twilio Message Service SID" you can use the Sender Phone Number gener
 
 Now the backend should be setup, we can proceed to add our client-side code!
 
+<Admonition type="info">
+  The `Twilio Content SID` field is specific to Twilio Programmable Messaging (WhatsApp) senders.
+  You may disregard this field if you are only sending SMS messages. Refer to the section on
```
Comment (saltcod) [inline apps/docs/pages/guides/auth/phone-login/twilio.mdx:119]: can remove "you may"
just:
"Disregard this field if..."

Comment (J0) [inline apps/docs/pages/guides/auth/phone-login/twilio.mdx:119]:
```suggestion
Disregard this field if you are only sending SMS messages. Refer to the section on
```

File: apps/docs/pages/guides/auth/phone-login/twilio.mdx (hunk 1 above)
Comment (J0) [inline apps/docs/pages/guides/auth/phone-login/twilio.mdx:414]:
```suggestion
Note that you may only use one Twilio Content SID at a time. Use Twilio Verify or the Basic Authentication Message if you require internationalization for your message template.
```

**Review by joshenlim — APPROVED** [review]

**Pull request merged**